### PR TITLE
PWN-6238 - send amount make field and layout

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6238

## Description of Work

PWN-6238 - make layout and make some preparations

## Screenshot
P2P Wallet – widget_send_details_input  P2P_Wallet ui-kit main  2022-12-06 16-42-19

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt [hunk 1]
```diff
@@ -33,9 +36,29 @@ class NewSendFragment :
...
+        }
+        // TODO PWN-6090 make button
+        binding.sliderSend.setActionText(R.string.send_enter_amount)
+    }
+
+    // TODO PWN-6090 connect this method to token selection
+    fun showSourceToken(token: Token.Active) {
+        with(binding.widgetSendDetails) {
+            glideManager.load(imageViewTokenIcon, token.iconUrl)
+            textViewTokenName.text = token.tokenSymbol
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt:59]: i suggest to do this as a widget method

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSendDetailsWidget.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding: WidgetSendDetailsInputBinding = inflateViewBinding()
+
+    var amountListener: ((input: String) -> Unit)? = null
+    val switchListener: (() -> Unit)? = null
+    val tokenClickListener: (() -> Unit)? = null
+
+    val imageViewTokenIcon
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSendDetailsWidget.kt:22]: this fields is redundant if you make just a method setToken

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSendDetailsWidget.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    }
+
+    override fun onDetachedFromWindow() {
+        AmountFractionTextWatcher.uninstallFrom(binding.editTextAmount)
+        super.onDetachedFromWindow()
+    }
+
+    fun setSwitchLabel(text: String) {
+        binding.textViewTokenSwitchLabel.text = text
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSendDetailsWidget.kt:58]: textViewAmountTypeSwitchLabel

we don't switch token here

File: app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt [hunk 4]
```diff
@@ -33,9 +36,29 @@ class NewSendFragment :
...
 
     override val presenter: NewSendContract.Presenter by inject()
+    private val glideManager: GlideManager by inject()
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding.toolbar.title = recipientUsername ?: recipientAddress
+        // TODO PWN-6090 make logic of setting token!
+        binding.widgetSendDetails.apply {
+            setSendLabel(getString(R.string.send_label))
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt:46]: always is `send_label`, i think no need to change it in future

**Review by gslevinkov — APPROVED** [review]

**Comment by gslevinkov** [comment]
thanks for the help!

File: app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt:59]: what do you mean widget method ?

File: app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt (hunk 4 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt:46]: yes but it is also not a UiKit part 

maybe we will need to make it with attrs

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt:46]: hmm maybe

ok I will move it to UiKit

**Comment by kabramovp2p** [comment]
> thanks for the help!

it is only the beginning :(

File: ui-kit/src/main/res/drawable/ic_arrows.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<vector
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply ui-kit/src/main/res/drawable/ic_arrows.xml:1]: image

everything looks fine ?

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSendDetailsWidget.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding: WidgetSendDetailsInputBinding = inflateViewBinding()
+
+    var amountListener: ((input: String) -> Unit)? = null
+    val switchListener: (() -> Unit)? = null
+    val tokenClickListener: (() -> Unit)? = null
+
+    val imageViewTokenIcon
+        get() = binding.imageViewTokenIcon
```
Comment (1sakov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSendDetailsWidget.kt:23]: The view should make its fields public. 
If we need to load images into it, let's inject GlideManager here and create a method: 

`fun loadImage(imageUrl: String)`

The only thing is to move Glide manager

**Review by gslevinkov — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

File: ui-kit/src/main/res/drawable/bg_rounded_solid_smoke_14.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+
+    <solid android:color="@color/bg_smoke" />
+    <corners android:radius="14dp" />
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply ui-kit/src/main/res/drawable/bg_rounded_solid_smoke_14.xml:7]: Send (Android) – Figma 2022-12-07 13-56-30

**Pull request merged**